apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,20 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgTemplateOutlet } from '@angular/common';
+import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
+import { ModelWithId } from '../../model/model-with-id.model';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  host: {
+    class: 'flex flex-col gap-3 p-4 border-2 border-black rounded-md w-fit',
+  },
+  imports: [NgFor, NgTemplateOutlet, ListItemComponent],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
+export class CardComponent<T extends ModelWithId> {
+  @Input() items: T[] | null = null;
 
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @ContentChild('listItemRef', { read: TemplateRef })
+  listItemRef!: TemplateRef<{ $implicit: T }>;
```

Comment:
very nice, but you can do better using a directive. Magic string is not always the better choice